sklearn/preprocessing/value_dropper.py

```diff
@@ -0,0 +1,216 @@
+        correlate with.
+
+        If this is specified n_labels argument is ignored.
+
+    missing_mask_only : bool, default False
+        Whether to modify/return only the missing mask without
+        touching the actual data X.
+
+    return_missing_mask : bool, default False
+        Whether to return the missing mask along with the data (X, y)
+
+    return_labels : bool, default False
+        Whether to return the picked labels
+
+    copy : bool, default False
+        Whether to copy the data (and missing_mask) or work inplace.
+
+    verbose : bool, default False
+        Whether to print the number of correlated/non-correlated missing values
+        that will be introduced.
+
+    random_state : int, optional
+        The seed for the numpy's random number generator.
+    Returns
+    -------
+
+    X, y (, missing_mask, labels) : Tuple
+        Returns missing_mask if return_missing_mask is set to True
+        Returns labels if return_labels is set to True
+
+    """
+    X, y = check_X_y(X, y)
+
+    if missing_mask_only and not return_missing_mask:
+        raise ValueError("Both missing_mask_only and return_missing_mask"
+                         "cannot be True")
+
+    if missing_fraction >= 1:
+        raise ValueError("The missing_fraction cannot be greater than"
+                         " or equal to 1.")
+
+    if copy:
+        X = X.copy()
+        if missing_mask is not None:
+            missing_mask = missing_mask.copy()
+
+    if (isinstance(missing_values, str) and
+            missing_values.lower() == "NaN"):
+        missing_values = np.nan
+
+    if missing_mask is None:
+        if np.isnan(missing_values):
+            missing_mask = np.isnan(X)
+        else:
+            missing_mask = X == missing_values
+
+    n_samples, n_features = X.shape
+    n_elements = n_samples * n_features
+
+    current_n_missing = np.count_nonzero(missing_mask)
+    required_n_missing = int(missing_fraction * n_elements)
+
+    if current_n_missing > required_n_missing:
+        raise ValueError("There are currently %d missing values, "
+                         "which is >= a fraction of %0.2f that is"
+                         "expected to be missing."
+                         % (current_n_missing, missing_fraction))
+
+    rng = check_random_state(random_state)
+    n_more_missing = required_n_missing - current_n_missing
+
+    unique_labels = np.unique(y)
+    n_unique_labels = len(unique_labels)
+
+    if labels is None:
+        # Labels is an int specifying the no of labels to correlate
+        # with
+
+        if n_labels > n_unique_labels:
+            raise ValueError("The n_labels (%d) is greater than"
+                             " no of unique labels in y (%d)"
+                             % (n_labels, n_unique_labels))
+
+        labels = rng.choice(n_unique_labels, n_labels, replace=False)
+        # Reset the RNG as we don't want this operation to affect
```

Comment:
Yes, there seems to be no benefit in this. Apart from which, it has no effect if someone passed a `np.random.RandomState` instance